assets/state-vfio-manager/0400_configmap.yaml

```diff
@@ -123,6 +125,35 @@ data:
         fi
     }
 
+    get_graphics_aux_dev() {
+        local gpu=$1
+        device_class_file=$(readlink -f "/sys/bus/pci/devices/$gpu/class")
+        device_class=$(cat "$device_class_file")
+        if [ "$device_class" == "0x030000" ]; then
```

Comment:
Some questions for my understanding:

Will there ever be a scenario where the GPU is configured in graphics mode, but no auxilary device is present? Also, does device class `0x03000` always indicate a GPU is in graphics mode?

Reply:
Per https://docs.nvidia.com/grid/13.0/grid-gpumodeswitch-user-guide/index.html#compute-and-graphics-mode, the answer to the second question is yes. Not sure about the first question. But the script will gracefully handle this case, by doing nothing.

Reply:
```
Classcode | VGA Controller | This classcode indicates to OS that the GPU can function as a primary display device.
```
I believe this answers my second question. Thanks.

> Not sure about the first question. But the script will gracefully handle this case, by doing nothing.

It is not clear to me that we handle this case gracefully. If we find a GPU that is a VGA controller (class 0x030000), we always return the output of `aux_dev=$(ls "/sys/bus/pci/devices/$gpu" | grep consumer | awk -Fconsumer:pci: '{print $2}')` regardless if there is a valid auxiliary device or not. We then call `bind_pci_device "$aux_dev"` with this string. I am not sure what the behavior of `bind_pci_device` is for invalid input.

Reply:
Right, let me push a fix to handle this case.

Reply:
I am curious how this will change once we enable SR-IOV and we have VFs. I haven't checked how the VFs are enumerated regarding BDF in graphics mode. Will the `consumer` link change or not, I suppose not because what `consumer` means is that a `pci` device with BDF xxxx:xx:xx.1 uses the parent device that we're currently inspecting. 
```
consumer:pci:0000:19:00.1 -> ../../../../../virtual/devlink/pci:0000:19:00.0--pci:0000:19:00.1
```

Reply:
Can we check that we have a BDF that is returned if anything is returned?

Reply:
Or do a check if we have a multi function device if we're not using sriov?

Reply:
Added a check to verify /sys/bus/pci/devices/$aux_dev/ exists.